x-pack/test/api_integration/apis/security_solution/matrix_dns_histogram.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import {
+  NetworkDnsEdges,
+  MatrixHistogramQuery,
+  MatrixHistogramType,
+} from '../../../../plugins/security_solution/common/search_strategy';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getService }: FtrProviderContext) {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertest');
+
+  describe('Matrix DNS Histogram', () => {
+    describe('Large data set', () => {
+      before(() => esArchiver.load('security_solution/matrix_dns_histogram/large_dns_query'));
+      after(() => esArchiver.unload('security_solution/matrix_dns_histogram/large_dns_query'));
+
+      const FROM = '2000-01-01T00:00:00.000Z';
+      const TO = '3000-01-01T00:00:00.000Z';
+
+      it('Make sure that we get dns data without getting bucket errors when querying large volume of data', async () => {
+        const { body: networkDns } = await supertest
+          .post('/internal/search/securitySolutionSearchStrategy/')
+          .set('kbn-xsrf', 'true')
+          .send({
+            defaultIndex: ['large_volume_dns_data'],
+            docValueFields: [],
+            factoryQueryType: MatrixHistogramQuery,
+            histogramType: MatrixHistogramType.dns,
+            filterQuery:
+              '{"bool":{"must":[],"filter":[{"match_all":{}}],"should":[],"must_not":[]}}',
+            isPtrIncluded: false,
+            timerange: {
+              interval: '12h',
+              to: TO,
+              from: FROM,
+            },
+            wait_for_completion_timeout: '10s',
```

Comment:
As an aside, I don't think when we add this `wait_for_completion_timeout` the enhanced data/async search actually honors it. Before I fixed the bug to I noticed that I would always get back a `isRunning` set to true and I had to manually change the wait for completion within Kibana core and then fail the test.

Luckily it runs fast enough here once fixed. However, I think this might be part of a source of why we are seeing some sporadic failures on CI. I added a `expect(networkDns.isRunning).to.equal(false);` below this in case this begins failing on CI we will know it is because the large query takes a while to run and the `wait_for_completion_timeout` is not being honored.

Reply:
Yeah, ideally wait_for_completion_timeout should avoid SearchStrategy giving us partial result and isRunning should always be false. Thanks for doing this, if this test fails then I can remove this option for all the tests.